File: python solutions/shortest_word_distance.py
from typing import List


class Solution:
    def shortestDistance(self, wordsDict: List[str], word1: str, word2: str) -> int:
        short_d = len(wordsDict)
        s = e = -1
        for i in range(len(wordsDict)):
            if wordsDict[i] == word1: s = i
            if wordsDict[i] == word2: e = i
            if s != -1 and e != -1: short_d = min(short_d, abs(s-e))
        return short_d

File: ruby solutions/shortest_word_distance.rb
# @param {String[]} words_dict
# @param {String} word1
# @param {String} word2
# @return {Integer}
def shortest_distance(words_dict, word1, word2)
    min_d = words_dict.length
    s = e = -1
    for i in 0...words_dict.length do
        s = i if words_dict[i] == word1
        e = i if words_dict[i] == word2
        min_d = [min_d, (s-e).abs].min if s != -1 and e != -1
    end
    min_d
end